sos/plugins/vdsm.py

```diff
@@ -0,0 +1,153 @@
+# Copyright (C) 2018 Red Hat, Inc.
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+
+import ast
+import os
+import stat
+import re
+
+from sos.plugins import Plugin, RedHatPlugin
+
+
+# This configuration is based on vdsm.storage.lvm.LVM_CONF_TEMPLATE.
+#
+# locking_type is set to 0 in order to match lvm sos commands. With this
+# configuration we don't take any locks, so we will never block because
+# there is a stuck lvm command.
+# locking_type=0
+#
+# use_lvmetad is set to 0 in order not to show cached, old lvm metadata.
+# use_lvmetad=0
+#
+# preferred_names and filter config values are set to capture Vdsm devices.
+# preferred_names=[ '^/dev/mapper/' ]
+# filter=[ 'a|^/dev/mapper/.*|', 'r|.*|' ]
+LVM_CONFIG = """
+global {
+    locking_type=0
+    use_lvmetad=0
+}
+devices {
+    preferred_names=[ "^/dev/mapper/" ]
+    ignore_suspended_devices=1
+    write_cache_state=0
+    disable_after_error_count=3
+    filter=[ "a|^/dev/mapper/.*|", "r|.*|" ]
+}
```

Comment:
Need to run this through the `pycodestyle` command - pep8 doesn't like space around `[]` and `()` members:

```
$ echo 'foo = [ "bar" ]' | pycodestyle -
stdin:1:8: E201 whitespace after '['
stdin:1:14: E202 whitespace before ']'
```

Reply:
pycodestyle didn't raise any error.

Reply:
Ah, right - it's inside a string, of course. I don't really care how you format external data in strings, beyond their own requirements (LVM2 config snippets in this case), although if it's not a big deal for you then it might help avoid other readers having a double take here (looking at the diff it's less clear that it's a string than seeing it in context).

Reply:
Ah, right - it's inside a string, of course. I don't really care how you format external data in strings, beyond their own requirements (LVM2 config snippets in this case), although if it's not a big deal for you then it might help avoid other readers having a double take here (looking at the diff it's less clear that it's a string than seeing it in context).

Reply:
Ah, right - it's inside a string, of course. I don't really care how you format external data in strings, beyond their own requirements (LVM2 config snippets in this case), although if it's not a big deal for you then it might help avoid other readers having a double take here (looking at the diff it's less clear that it's a string than seeing it in context).

Reply:
Ah, right - it's inside a string, of course. I don't really care how you format external data in strings, beyond their own requirements (LVM2 config snippets in this case), although if it's not a big deal for you then it might help avoid other readers having a double take here (looking at the diff it's less clear that it's a string than seeing it in context).

Reply:
Ah, right - it's inside a string, of course. I don't really care how you format external data in strings, beyond their own requirements (LVM2 config snippets in this case), although if it's not a big deal for you then it might help avoid other readers having a double take here (looking at the diff it's less clear that it's a string than seeing it in context).

Reply:
Ah, right - it's inside a string, of course. I don't really care how you format external data in strings, beyond their own requirements (LVM2 config snippets in this case), although if it's not a big deal for you then it might help avoid other readers having a double take here (looking at the diff it's less clear that it's a string than seeing it in context).

Reply:
Ah, right - it's inside a string, of course. I don't really care how you format external data in strings, beyond their own requirements (LVM2 config snippets in this case), although if it's not a big deal for you then it might help avoid other readers having a double take here (looking at the diff it's less clear that it's a string than seeing it in context).

Reply:
Ah, right - it's inside a string, of course. I don't really care how you format external data in strings, beyond their own requirements (LVM2 config snippets in this case), although if it's not a big deal for you then it might help avoid other readers having a double take here (looking at the diff it's less clear that it's a string than seeing it in context).

Reply:
Ah, right - it's inside a string, of course. I don't really care how you format external data in strings, beyond their own requirements (LVM2 config snippets in this case), although if it's not a big deal for you then it might help avoid other readers having a double take here (looking at the diff it's less clear that it's a string than seeing it in context).

Reply:
I removed the spaces

Reply:
:)